packages/blade/src/utils/isPartialMatchObjectKeys/isPartialMatchObjectKeys.ts

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable no-console */
+import isObject from 'lodash/isObject';
+
+export type DeepPartial<T> = {
+  [P in keyof T]?: T[P] extends Record<number | string, unknown> ? DeepPartial<T[P]> : T[P];
+};
+
+const isPartialMatchObjectKeys = <ActualObject>({
+  objectToMatch,
+  objectToInspect,
+}: {
+  objectToMatch: DeepPartial<ActualObject>;
+  objectToInspect: ActualObject;
+}): boolean => {
+  const matchResponses: boolean[] = [];
+
+  const matchObjectKeys = ({
+    innerObjectToMatch,
+    innerObjectToInspect,
+  }: {
+    innerObjectToMatch: DeepPartial<ActualObject>;
+    innerObjectToInspect: DeepPartial<ActualObject>;
+  }): void => {
+    for (const [key, valueToMatch] of Object.entries(innerObjectToMatch)) {
+      const valueToInspect = innerObjectToInspect[key as keyof DeepPartial<ActualObject>];
+
+      if (innerObjectToInspect.hasOwnProperty(key)) {
+        if (
+          valueToMatch === null ||
+          valueToMatch === undefined ||
+          valueToMatch === '' ||
+          Array.isArray(valueToMatch) ||
+          // the condition checks if the "valueToMatch" is not of type object then "valueToMatch" type should be same as type of "valueToInspect"
+          (!(valueToMatch instanceof Object) && typeof valueToMatch !== typeof valueToInspect)
+        ) {
+          // this is an invalid case, so we log error
+          console.error(
+            `[isPartialMatchObjectKeys]: Unexpected value: ${JSON.stringify(
+              valueToMatch,
+            )} of type ${typeof valueToMatch} for key: ${key}`,
+          );
+          matchResponses.push(false);
+        }
+
+        if (typeof valueToMatch === 'string') {
+          // we have reached leaf node of the objectToMatch(i.e sub-object)
+          matchResponses.push(true);
```

Comment:
No need to match or equate `valueToMatch` and `valueToInspect`?

Reply:
All the checks have been done above and hence it's reaching here so doesn't makes sense to do it again

Reply:
because typeof is not straightforward as our human brains think. typeof null, undefined all leads to object so `valueToMatch: null === valueToInspect: {}` hence there's a place where I explicitly check for `isObject(valueToMatch) && isObject(valueToInspect)`